src/v/datalake/backlog_controller.cc

```diff
@@ -51,22 +57,29 @@ ss::future<> backlog_controller::stop() {
 
 void backlog_controller::update() {
     using namespace std::chrono_literals;
-
+    auto now = std::chrono::steady_clock::now();
     _current_sample = _sampling_f();
 
     auto current_err = _current_sample - _setpoint();
-    auto update = _proportional_coeff() * current_err;
+    _total_err += (current_err / ((now - _last_sample_time) / 1ms));
```

Comment:
Probably a bit late, as we're about to release, so rather as an idea. Total integral error will be affected by figures from distant past. Would it make sense for it to slowly decay as we accumulate it, exponentially with time?